### PR TITLE
CI: MQ sync status check fix

### PR DESCRIPTION
**Pull request by maxknv**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [x]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 0d2e0e3131b9d97f03a13d9e7bb2f7208262d410 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62591/0d2e0e3131b9d97f03a13d9e7bb2f7208262d410/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/ci/sync_pr.py [hunk 1]
```diff
@@ -53,12 +54,36 @@ def merge_sync_pr(gh, sync_pr):
...
-            get_commit(gh, pr_info.sha), FAILURE, "", "Sync PR failure", "A Sync"
+            get_commit(gh, pr_info.sha), FAILURE, "", "Sync PR not found", "A Sync"
         )
-    else:
+        return
+
+    retries = 0
+    while sync_pr.mergeable_state == "unknown" and retries < 3:
+        retries += 1
+        print(f"Unknown status. Trying to fetch again [{retries}/3]")
```
Comment (Felixoid) [inline tests/ci/sync_pr.py:66]: It's a good opportunity to get rid of prints. Let's get used to `logging`

**Review by Felixoid — CHANGES_REQUESTED** [review]

File: tests/ci/sync_pr.py [hunk 2]
```diff
@@ -53,12 +54,36 @@ def merge_sync_pr(gh, sync_pr):
...
+        time.sleep(5)
+        sync_pr = gh.get_pulls_from_search(
```
Comment (Felixoid) [inline tests/ci/sync_pr.py:69]: This is wrong. You need to use `repo.get_pull(number)`

File: tests/ci/sync_pr.py (hunk 1 above)
Comment (maxknv) [reply tests/ci/sync_pr.py:67]: print does all that we need here

File: tests/ci/sync_pr.py [hunk 3]
```diff
@@ -53,12 +54,37 @@ def merge_sync_pr(gh, sync_pr):
 
 
 def set_sync_status(gh, pr_info, sync_pr):
```
Comment (Felixoid) [inline tests/ci/sync_pr.py:56]: Adding a proper typing here would help us to prevent issues like in line 69

**Review by Felixoid — APPROVED** [review]

**Pull request merged**